brave/src/main/java/brave/secondary_sampling/SecondarySampler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.secondary_sampling;
+
+import brave.http.HttpServerRequest;
+import brave.propagation.TraceContext;
+
+public interface SecondarySampler {
+  /**
+   * Returns a trigger function that decides if this request is sampled for the given {@code
+   * samplingKey}.
```

Comment:
If the `samplingKey` is created and given, it means the request is already sampled and passing a request object or grpc server call object again is redundant and might not be needed.

The request object or the grpc server call object will be needed in the place where the `sampling_key` is set initially where the trace is started.

Reply:
secondary sampling is a local decision, which means you need to re-interpret the decision at every hop, not at the beginning of a trace. This is the fundamental difference between this and primary sampling, which is done at the beginning of the request. I'll update the javadoc.